File: pl/0603.pl
use Math::Trig 'pi';

my $a = 0;
my $b = 1;

# Simpson‘¥‚ÅÏ•ª
my $n = 2;
for $j (1..5)
{
    my $h  = ($b - $a) / $n;
    my $s2 = 0;
    my $s4 = 0;
    my $x  = $a + $h;
    for $i (1..($n / 2))
    {
        $s4 += f($x);
        $x  += $h;
        $s2 += f($x);
        $x  += $h;
    }
    $s2 = ($s2 - f($b)) * 2 + f($a) + f($b);
    $s4 *= 4;
    my $s = ($s2 + $s4) * $h / 3;
    $n *= 2;

    # Œ‹‰Ê‚ğ ƒÎ ‚Æ”äŠr
    printf("%2d : %13.10f, %13.10f\n", $j, $s, $s - pi);
}

sub f
{
    my ($x) = @_;
    4 / (1 + $x * $x);
}

File: js/0603.js
var a = 0
var b = 1

// Simpson‘¥‚ÅÏ•ª
var n = 2
for (var j = 1; j <= 5; j++)
{
    var h  = (b - a) / n 
    var s2 = 0
    var s4 = 0
    var x  = a + h
    for (var i = 1; i <= n / 2; i++)
    {
        s4 += f(x)
        x  += h
        s2 += f(x)
        x  += h
    }
    s2 = (s2 - f(b)) * 2 + f(a) + f(b)
    s4 *= 4
    var s = (s2 + s4) * h / 3
    n *= 2

    // Œ‹‰Ê‚ğ ƒÎ ‚Æ”äŠr
    WScript.StdOut.Write(("  "            + j                        ).slice( -2) + " : ");
    WScript.StdOut.Write(("             " + s.toFixed(10)            ).slice(-13) + ", ");
    WScript.StdOut.Write(("             " + (s - Math.PI).toFixed(10)).slice(-13) + "\n" );
}

function f(x)
{
    return 4 / (1 + x * x)
}
